contracts/contracts/TrustlinesNetworkToken.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity ^0.5.8;
+
+import "./SafeMath.sol";
+
+
+contract TrustlinesNetworkToken {
+
+    using SafeMath for uint256;
+
+    uint constant MAX_UINT = 2**256 - 1;
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+    uint256 private _totalSupply;
+
+    mapping (address => uint256) private _balances;
+    mapping (address => mapping (address => uint256)) private _allowances;
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+
+    constructor (string memory name, string memory symbol, uint8 decimals, address preMintAddress, uint256 preMintAmount) public {
+        _name = name;
+        _symbol = symbol;
+        _decimals = decimals;
+
+        _mint(preMintAddress, preMintAmount);
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        return _balances[account];
+    }
+
+    function allowance(address owner, address spender) public view returns (uint256) {
+        return _allowances[owner][spender];
+    }
+
+    function name() public view returns (string memory) {
+        return _name;
+    }
+
+    function symbol() public view returns (string memory) {
+        return _symbol;
+    }
+
+    function decimals() public view returns (uint8) {
+        return _decimals;
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    function transfer(address recipient, uint256 amount) public returns (bool) {
+        _transfer(msg.sender, recipient, amount);
+    }
+
+    function approve(address spender, uint256 value) public returns (bool) {
+        require(value == 0 || _allowances[msg.sender][spender] == 0, "ERC20: approve only to or from 0 value");
+        _approve(msg.sender, spender, value);
+    }
+
+    function transferFrom(address sender, address recipient, uint256 amount) public returns (bool) {
+        _transfer(sender, recipient, amount);
+
+        uint allowance = _allowances[sender][msg.sender];
+        uint updatedAllowance = allowance.sub(amount);
+        if (allowance < MAX_UINT) {
+            _approve(sender, msg.sender, updatedAllowance);
```

Comment:
why do you reuse `_approve` here. This seems wrong, as you dont approve again here, so just substract from the allowance.

Reply:
That's just how they do it in the original: https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC20/ERC20.sol#L96
I'm fine with changing it, I just changed the minimum I could not to introduce complexity / bugs.

Reply:
But I also want to throw an approve event, otherwise we cannot track the change of allowance through events.

Reply:
I am fine with it